github-org-scan-docker.groovy

```diff
@@ -1,227 +1,335 @@
+                    }
+                }
             }
-            parallel targets
-          }
         }
-      }
     }
-  }
 }
 
 /**
  * Generate Scan Stages
- * 
+ *
  * This function generates Jenkins pipeline stages for running `endorctl` scans on a specified set of targets.
  * Each stage represents a scan job for a specific project within a Jenkins pipeline.
- * 
+ *
  * @param targets (Map<String, Map>): A map where the keys are project names and the values are maps
  *   containing the configuration for each project.
- * 
+ *
  * @param project (String): The URL of the target repository or project to scan.
- * 
+ *
  * @param args (Map<String, String>): Arguments to pass to the Jenkins pipeline stages.
- * 
+ *
  * @return (Map<String, Map>): A map representing the generated Jenkins pipeline stages.
  *   Each stage corresponds to a project and includes the steps required to scan the project.
  */
 
 def generate_scan_stages(def targets, def project, def args) {
-  if (project) {
-    targets[project] = {
-      def DockerScan = new DockerScan()
-      def Checkout = new Checkout()
-      String projectName = projectName(project)
-      String stageName = "Scan " + projectName
-      node(args['AGENT_LABEL']) {
-        stage(stageName) {
-          try {
-            String workspace = Checkout.getWorkSpace(this, project)
-            Checkout.setCredentialHelper(this)
-            Checkout.clone(this, args, project, workspace)
-            def branch = Checkout.getDefaultBranch(this, project, workspace)
-            Checkout.execute(this, branch, workspace)
-            DockerScan.execute(this, args, project, branch, workspace)
-          } catch (err) {
-            echo err.toString()
-            unstable("endorctl Scan failed for ${project}")
-          }
+    if (project) {
+        targets[project] = {
+            def DockerScan = new DockerScan()
+            def Checkout = new Checkout()
+            String projectName = projectName(project)
+            String stageName = "Scan " + projectName
+            node(args['AGENT_LABEL']) {
+                stage(stageName) {
+                    try {
+                        String workspace = Checkout.getWorkSpace(this, project)
+                        Checkout.setCredentialHelper(this)
+                        Checkout.clone(this, args, project, workspace, false)
+                        def branch = Checkout.getDefaultBranch(this, project, workspace)
+                        Checkout.execute(this, branch, workspace)
+                        DockerScan.execute(this, args, project, branch, workspace)
+                    } catch (err) {
+                        echo err.toString()
+                        unstable("endorctl Scan failed for ${project}")
+                    }
+                }
+            }
+        }
+    }
+}
+
+def filterProjects(def pipeline, def projects, def args, def projectsWithUUID) {
+    def scannableProjects = []
+    setGHCreds(this)
```

Comment:
Not sure if this function needs to be called here or after line number 175, just invoke 
`Checkout.setCredentialHelper(this)`

The setGHCreds is not called anywhere else.

Reply:
The func `projectHasCommitsWithinLastNDays` is being called in loop for each project at line#147 , and essentially the func `setGHCreds` is setting up the gh creds in the underlying machine so setting this only once at line#140 is logical.